src/sampling/adaptive.jl

```diff
@@ -0,0 +1,204 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+using DataStructures: BinaryHeap
+using LinearAlgebra: norm
+import IterTools
```

Comment:
The only place in the source code where we import modules is the main Meshes.jl file. That way we don't loose track of current imports and avoid naming conflicts.